app/src/ble.c

```diff
@@ -137,7 +359,52 @@ static void auth_cancel(struct bt_conn *conn)
     LOG_DBG("Pairing cancelled: %s", log_strdup(addr));
 }
 
+#if !IS_ENABLED(CONFIG_ZMK_SPLIT_BLE_ROLE_PERIPHERAL)
+static enum bt_security_err auth_pairing_accept(struct bt_conn *conn, const struct bt_conn_pairing_feat *const feat)
+{
+    struct bt_conn_info info;
+    bt_conn_get_info(conn, &info);
+
+    LOG_DBG("role %d, open? %s", info.role, active_profile_is_open() ? "yes" : "no");
+    if (info.role != BT_CONN_ROLE_SLAVE && !active_profile_is_open()) {
```

Comment:
```suggestion
    if (info.role == BT_CONN_ROLE_SLAVE && !active_profile_is_open()) {
```
As discussed in Discord. This should check whether this is a connection to a host. In that case the central side has the role `BT_CONN_ROLE_SLAVE`, so it should be `==`. This is confirmed working on split and non-split on my end.

Reply:
Fixed.